instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -99,8 +102,48 @@ public static boolean isRegistered() {
     return REGISTERED.get();
   }
 
-  private static Driver findDriver(String realUrl) {
-    for (Driver candidate : Collections.list(DriverManager.getDrivers())) {
+  /**
+   * Add driver candidate that wasn't registered against {@link DriverManager}. This is mainly
+   * useful when drivers are initialized at build time, and you don't want to postpone class
+   * initialization (f.e. runtime initialization could make driver incompatible with other systems).
```

Comment:
Hey @trask ,

Hard to make generalizations for building native images, it depends on you use other systems, if you have everything runtime initialized, of course there is no issue, but that's very much against what we want (with GraalVM native images, it's desirable that you do all you can at build time). Having runtime initialized Oracle driver made it incompatible with DB2 driver as it made it incompatible with the JDK DriverManager integrations. DriverManager will typically need to load all drivers in a different phase. Reason for this PR is that registering driver again static global registry DriverManager is not pattern I'd like to follow. F.e. using Agroal connection pool, you don't need to have drivers registered against DriverManager and I don't see why should OpenTelemetryDriver require it for underlying drivers. Why should driver make changes outside its scope (java.sql) by requiring runtime initialization? Though IMO it's correct it register itself.

In our case, if we had Oracle drive initialized at runtime it would not compile when used with other JDBC drivers, as other implementations need this class initialized during build.